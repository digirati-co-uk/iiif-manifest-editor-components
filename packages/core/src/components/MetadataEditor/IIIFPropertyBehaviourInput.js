import * as React from 'react';
import * as PropTypes from 'prop-types';
import {
  FormControl,
  FormLabel,
} from '@material-ui/core';

import IIIFBehaviours from '../fields/IIIFBehaviours/IIIFBehaviours';

const IIIFPropertyBehaviourInput = ({
  labels,
  update,
  targetEntity,
  lang,
  classes,
  type
}) => (
  <FormControl component="fieldset">
    <FormLabel component="legend">
      {labels[`${type}.Behaviors`] || 'Behaviors'}
    </FormLabel>
    <IIIFBehaviours
      classes={classes}
      target={targetEntity}
      lang={lang}
      update={update}
      labels={labels}
    />
  </FormControl>
)

IIIFPropertyBehaviourInput.propTypes ={
  labels: PropTypes.object,
  update: PropTypes.func,
  targetEntity: PropTypes.object,
  lang: PropTypes.string,
  classes: PropTypes.any,
  type: PropTypes.string,
};

export default IIIFPropertyBehaviourInput;