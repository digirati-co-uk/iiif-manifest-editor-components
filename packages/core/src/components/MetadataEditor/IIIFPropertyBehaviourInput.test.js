import * as React from 'react';
import IIIFPropertyBehaviourInput from './IIIFPropertyBehaviourInput';

describe("IIIFPropertyBehaviourInput", () => {
  it('Renders IIIFPropertyBehaviourInput', () => {
    const handleUpdate = jest.fn();
    const target = {
      behavior: [],
    }
    const languageCode = 'en';
    const type = 'Manifest';
    const labels = {};
    const classes = {};
    const renderedInput = render(
      <IIIFPropertyBehaviourInput
        classes={classes}
        labels={labels}
        update={handleUpdate}
        targetEntity={target}
        lang={languageCode}
        type={type}
      />
    );
    expect(renderedInput).toMatchSnapshot();
  });
  //TODO: on change????
});