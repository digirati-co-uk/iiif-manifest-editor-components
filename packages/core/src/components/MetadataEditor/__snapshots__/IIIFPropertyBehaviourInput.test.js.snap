// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IIIFPropertyBehaviourInput Renders IIIFPropertyBehaviourInput 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiFormControl-root-1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiFormLabel-root-5",
        },
        "children": Array [
          Object {
            "data": "Behaviors",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "legend",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiFormControl-root-1 MuiFormControl-marginDense-3",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MuiInputBase-root-25 MuiOutlinedInput-root-12 MuiInputBase-formControl-26 MuiInputBase-marginDense-32",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "MuiPrivateNotchedOutline-root-42 MuiOutlinedInput-notchedOutline-19",
                    "style": "padding-left:8px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "MuiPrivateNotchedOutline-legend-43",
                        "style": "width:0.01px",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "​",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                      ],
                      "name": "legend",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "fieldset",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "aria-invalid": "false",
                      "class": "MuiInputBase-input-35 MuiOutlinedInput-input-20 MuiInputBase-inputMarginDense-36 MuiOutlinedInput-inputMarginDense-21",
                      "type": "text",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-invalid": undefined,
                      "class": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-invalid": undefined,
                      "class": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "style": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "aria-invalid": "false",
                    "class": "MuiInputBase-input-35 MuiOutlinedInput-input-20 MuiInputBase-inputMarginDense-36 MuiOutlinedInput-inputMarginDense-21",
                    "type": "text",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "MuiPrivateNotchedOutline-root-42 MuiOutlinedInput-notchedOutline-19",
                      "style": "padding-left:8px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "MuiPrivateNotchedOutline-legend-43",
                          "style": "width:0.01px",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "data": "​",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                        ],
                        "name": "legend",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "fieldset",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-invalid": undefined,
                    "class": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-invalid": undefined,
                    "class": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiFormControl-root-1 MuiFormControl-marginDense-3",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiInputBase-root-25 MuiOutlinedInput-root-12 MuiInputBase-formControl-26 MuiInputBase-marginDense-32",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "MuiPrivateNotchedOutline-root-42 MuiOutlinedInput-notchedOutline-19",
                  "style": "padding-left:8px",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "MuiPrivateNotchedOutline-legend-43",
                      "style": "width:0.01px",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "​",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                    ],
                    "name": "legend",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "fieldset",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "aria-invalid": "false",
                    "class": "MuiInputBase-input-35 MuiOutlinedInput-input-20 MuiInputBase-inputMarginDense-36 MuiOutlinedInput-inputMarginDense-21",
                    "type": "text",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-invalid": undefined,
                    "class": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-invalid": undefined,
                    "class": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "style": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "aria-invalid": "false",
                  "class": "MuiInputBase-input-35 MuiOutlinedInput-input-20 MuiInputBase-inputMarginDense-36 MuiOutlinedInput-inputMarginDense-21",
                  "type": "text",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "MuiPrivateNotchedOutline-root-42 MuiOutlinedInput-notchedOutline-19",
                    "style": "padding-left:8px",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "MuiPrivateNotchedOutline-legend-43",
                        "style": "width:0.01px",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "​",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                      ],
                      "name": "legend",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "fieldset",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "style": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-invalid": undefined,
                  "class": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-invalid": undefined,
                  "class": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MuiFormLabel-root-5",
          },
          "children": Array [
            Object {
              "data": "Behaviors",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "legend",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "fieldset",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
